extra/import_data.py

```diff
@@ -0,0 +1,173 @@
+    # if latest date does not retrieve any analyses from Alissa, quit program and try later
+    if analysis is None:
+        logger.info('no new analysis in Alissa since last upload to VUSualizer')
+    # after uploading all analyses to MongoDB, replace time of last upload with starttime of script (before calling Alissa)
+    db.replace_one({"version": 1}, {"lastUpdatedOn": start_time, "version": 1}, True)
+
+
+def upload_to_mongodb(inheritance_analysis, accession_number, analyis_sources, patient_dn_no, vus_export, logger):
+    '''Function, for parsing data to the MongoDB'''
+    # connection with MongoDB and the correct database "vus" and collection "variant"
+    mongo_client = pymongo.MongoClient(config.mongodb_localhost)
+    db = mongo_client.vus.variant
+
+    # if DNnr already in mongoDB, check if updated an replace if neccesary
+    last_updated_on_mongoDB = db.find_one({"dn_no": patient_dn_no}, {"lastUpdatedOn": 1, "_id": 0})
+    if last_updated_on_mongoDB:
+        logger.info('dn_no: %s already present' % patient_dn_no)
+        last_updated_on_mongoDB = last_updated_on_mongoDB['lastUpdatedOn']
+        last_updated_on_Alissa = inheritance_analysis['lastUpdatedOn']
+        if last_updated_on_Alissa == last_updated_on_mongoDB:
+            logger.info('%s already in database, lastUpdatedOn Alissa is the same' % patient_dn_no)
+            return
+        elif last_updated_on_Alissa > last_updated_on_mongoDB:
+            db.delete_many({"dn_no": patient_dn_no})
+            logger.info('removed: %s from database, start replacing with newer version' % patient_dn_no)
+        elif last_updated_on_Alissa < last_updated_on_mongoDB:
+            logger.info('lastUpdatedOn older than within the database for %s, this should not happen' % patient_dn_no)
+            # TODO send (email) notification of this error
+
+    # get all relevant info from one patient into one dictionary.
+    patient = {}
+    patient["dn_no"] = patient_dn_no
+    logger.info('start uploading to mongodb: %s' % patient_dn_no)
+    patient["patient_accession_no"] = accession_number
+    patient.update(inheritance_analysis)
+    patient["analysis"] = patient.pop("reference")
+    patient["platform_dataset"] = analyis_sources["platformDataSet"]["info"]
+
+    # changes the format of the annotation sources from Alissa, to fit in MongoDB and resemble the old O-schijf format
+    externalSources_dict = {}
+    for source in analyis_sources['annotationSources']:
+        source_name = ""
+        source_value = ""
+        for key, value in source.items():
+            if key == 'name':
+                source_name = value
+            elif key == 'description':
+                source_value = value
+        if source_name and source_value:
+            externalSources_dict[source_name] = source_value
+        else:
+            logger.info("error, externalsources has changed format in Alissa")
```

Comment:
Is it possible to change the logger level to error?

Reply:
yes I already did this, based on your earlier comment about this 👍